### PR TITLE
GA Migration Java PostCommit Nexmark Dataflow 

### PR DESCRIPTION
**Pull request by andoni-guzman**
This PR is part of the migration from jenkins to github actions.

This PR contains this tests

Java PostCommit Nexmark Dataflow - Java 8 
Java PostCommit Nexmark Dataflow Direct - Java 8, Java 11 and  Java 17

2 Files were added to the workflows folder of github actions
     - job-postcommit-java-nexmark-dataflow and 
     - job-postcommit-java-nexmark-dataflow V2
4 Files were deleted, this files were the groovy files for jenkins
    - Dataflow
    - Dataflow V2 Java 8
    - Dataflow V2 Java 11
    - Dataflow V2 Java 17
Jenkins readme was modified deleting migrated tests.
CI.md was modified adding the new GA workflows 

Waiting for this PR before merge https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-postcommit-java-nexmark-dataflowV2.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  EXPORTSUMMARYTOBIGQUERY: true
+  TEMPLOCATION: gs://temp-storage-for-perf-tests/nexmark
+  INFLUXDATABASE: beam_test_metrics
+  INFLUXHOST: http://10.128.0.96:8086 
+  BASEINFLUXMEASUREMENT: nexmark
+  EXPORTSUMMARYTOINFLUXDB: true
+  INFLUXRETENTIONPOLICY: forever
+  INFLUXTAGSJAVA8: '{"runnerVersion":"V2","javaVersion":"8"}'
+  INFLUXTAGSJAVA11: '{"runnerVersion":"V2","javaVersion":"11"}'
+  INFLUXTAGSJAVA17: '{"runnerVersion":"V2","javaVersion":"17"}'
```
Comment (andoni-guzman) [inline .github/workflows/job-postcommit-java-nexmark-dataflowV2.yml:44]: The reason of having Influx tags in this way instead a matrix. is related a parse error using matrix and passing it into the gradlew command.

**Comment by fernando-wizeline** [comment]
LGTM!

File: .github/workflows/job-postcommit-java-nexmark-dataflow.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        querytype: ["sql","zetasql"]
+    timeout-minutes: 240
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - name: Setup environment
+        uses: ./.github/actions/setup-self-hosted-action
+      - name: Run *** RUN NEXMARK IN BATCH ${{matrix.streaming}} MODE USING DATAFLOW RUNNER ***
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-nexmark-dataflow.yml:58]:
```suggestion
      - name: Run Nexmark with Batch=${{matrix.streaming}} on Dataflow
```

Comment (damccorm) [reply .github/workflows/job-postcommit-java-nexmark-dataflow.yml:58]: Same comments elsewhere - please remove the bold/caps

**Review by damccorm — COMMENTED** [review]
Other than naming, this LGTM

File: .github/workflows/job-postcommit-java-nexmark-dataflow.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  BIGQUERYDATASET: nexmark
+  PROJECT: apache-beam-testing
+  RESOURCENAMEMODE: QUERY_RUNNER_AND_MODE
+  EXPORTSUMMARYTOBIGQUERY: true
+  TEMPLOCATION: gs://temp-storage-for-perf-tests/nexmark
+  INFLUXDATABASE: beam_test_metrics
+  INFLUXHOST: http://10.128.0.96:8086 
+  BASEINFLUXMEASUREMENT: nexmark
+  EXPORTSUMMARYTOINFLUXDB: true
+  INFLUXRETENTIONPOLICY: forever
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-nexmark-dataflow.yml:39]: These environment variables are used in a number of workflows. Should we refactor them to our shared variable config?

File: .github/workflows/job-postcommit-java-nexmark-dataflow.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+jobs:
+  java-postcommit-nexmark-dataflow:
+    name: Run PostCommit Java Nexmark - Dataflow 
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix:
+        streaming: ["true","false"]
+        querytype: ["sql","zetasql"]
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-nexmark-dataflow.yml:49]: See comment here - https://github.com/apache/beam/pull/23112#discussion_r995909354 - can we consolidate all these workflows to a single workflow that creates a job per config type?

**Comment by andoni-guzman** [comment]
Hi @damccorm .
I did some changes. I test all the changes to verify that the test were not broken. 
- I modify the names of the test like you suggest, like "Run Nexmark with Batch on Dataflow"
- In test-properties.json added all the variables that were repeated in the tests.
- In job-postcommit-java-nexmark-dataflow, use a matrix for all the tipes of executions of the job, only 1 job is in code, but in the workflow 6 jobs are executed
- In job-postcommit-java-nexmark-dataflowV2, some variables that were used in all the tests are getting from test-properties.json, the variables for Java 8, Java 11 and Java 17 still are in env variables because json parsing erros in other way.
Any other change please let me know

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**